build-selftests/action.yml

```diff
@@ -1,34 +1,27 @@
 name: 'build selftests'
 description: 'Build BPF selftests'
 inputs:
-  kernel:
-    description: 'kernel version'
-    default: 'LATEST'
-  toolchain:
-    description: 'what toolchain to use'
-    default: 'gcc'
-  kbuild-output:
-    description: 'relative or absolute path to use for storing build artifacts'
+  arch:
+    description: 'arch'
+    required: true
+  kernel-root:
+    description: 'Path to the root of the kernel source tree'
     required: true
-  max-make-jobs:
-    description: 'Maximum number of jobs to use when running make (e.g argument to -j). Default: 4*nproc'
-    default: ''
   llvm-version:
     description: 'llvm version'
     required: false
-    default: '16'
-  arch:
-    description: 'arch'
-    required: true
+    default: '18'
+  toolchain:
+    description: 'gcc or llvm'
+    default: 'gcc'
 
 runs:
   using: "composite"
   steps:
     - name: build selftests
       shell: bash
-      run: |
-        kbuild_output="$(realpath ${{ inputs.kbuild-output }})"
-        export LLVM_VERSION=${{ inputs.llvm-version }}
-        ${GITHUB_ACTION_PATH}/build_selftests.sh "${{ inputs.arch }}" "${{ inputs.kernel }}" "${{ inputs.toolchain }}" "${kbuild_output}"
       env:
-        MAX_MAKE_JOBS: ${{ inputs.max-make-jobs }}
```

Comment:
this used to be used to control how many cpus were used when compiling. On bare metal, with 96 CPUs, if all containers are building at the same time, they will all use that many CPUs, bringing the machine to a crawl.

Reply:
It's still used, it's just not an action input anymore. A caller can set it [like here](https://github.com/libbpf/libbpf/pull/862/files#diff-2ce184c8a6b8fb0bcdaaeb247ecef1e9d8e3badefc51505c2e5e731914cc93baR104).

The default is 4*nproc, [controlled here](https://github.com/libbpf/ci/blob/main/helpers.sh#L51-L57).